src/engine/engineworkerscheduler.cpp

```diff
@@ -12,13 +12,16 @@ EngineWorkerScheduler::EngineWorkerScheduler(QObject* pParent)
 }
 
 EngineWorkerScheduler::~EngineWorkerScheduler() {
+    // tell run method to terminate
+    const auto lock = lockMutex(&m_mutex);
     m_bQuit = true;
     m_waitCondition.wakeAll();
+    // wait for thread to terminate
```

Comment:
What is this `m_mutex` used for? Is there a risk of deadlock when `run` finishes to proceed? It might be safer to just keep `m_bQuit` in scope protection
```suggestion
    {
        // tell run method to terminate
        const auto lock = lockMutex(&m_mutex);
        m_bQuit = true;
        m_waitCondition.wakeAll();
    }
    // wait for thread to terminate
```

Reply:
The mutex was originally used to protected the m_pWorkers vector, are strangely enough only half to protect m_bQuit (protected in line 53 of the original code, but not in line 42 and not in line 17). I now protect it everywhere.

I don't think there is a risk of a race condition, but I agree with your change, as the scope of the mutex is indeed limited.